azurerm/internal/services/mssql/mssql_server_transparent_data_encryption_resource.go

```diff
@@ -0,0 +1,244 @@
+			return fmt.Errorf("Unable to parse key: %q: %+v", keyVaultKeyId, err)
+		}
+
+		// Make sure it's a key, if not, throw an error
+		if keyId.NestedItemType == "keys" {
+			keyName := keyId.Name
+			keyVersion := keyId.Version
+
+			// Extract the vault name from the keyvault base url
+			idURL, err := url.ParseRequestURI(keyId.KeyVaultBaseUrl)
+
+			if err != nil {
+				return fmt.Errorf("Unable to parse key vault hostname: %s", keyId.KeyVaultBaseUrl)
+			}
+
+			hostParts := strings.Split(idURL.Host, ".")
+			vaultName := hostParts[0]
+
+			// Create the key path for the Encryption Protector. Format is: {vaultname}_{key}_{key_version}
+			serverKeyName = fmt.Sprintf("%s_%s_%s", vaultName, keyName, keyVersion)
+		} else {
+			return fmt.Errorf("Key vault key id must be a reference to a key, but got: %s", keyId.NestedItemType)
+		}
+	}
+
+	// Service managed doesn't require a key name
+	encryptionProtectorProperties := sql.EncryptionProtectorProperties{
+		ServerKeyType: serverKeyType,
+		ServerKeyName: &serverKeyName,
+	}
+
+	// Only create a server key if the properties have been set
+	if serverKey.ServerKeyProperties != nil {
+		// Create a key on the server
+		futureServers, err := serverKeysClient.CreateOrUpdate(ctx, serverId.ResourceGroup, serverId.Name, serverKeyName, serverKey)
+		if err != nil {
+			return err
+		}
+
+		if err = futureServers.WaitForCompletionRef(ctx, serverKeysClient.Client); err != nil {
+			return err
+		}
+	}
+
+	encryptionProtectorObject := sql.EncryptionProtector{
+		EncryptionProtectorProperties: &encryptionProtectorProperties,
+	}
+
+	futureEncryptionProtector, err := encryptionProtectorClient.CreateOrUpdate(ctx, serverId.ResourceGroup, serverId.Name, encryptionProtectorObject)
+	if err != nil {
+		return err
+	}
+
+	if err = futureEncryptionProtector.WaitForCompletionRef(ctx, encryptionProtectorClient.Client); err != nil {
+		return fmt.Errorf("waiting on create/update future for Encryption Protector on server %q (Resource Group %q): %+v", serverId.Name, serverId.ResourceGroup, err)
+	}
+
+	// Encryption protector always uses "current" for the name
+	id := parse.NewEncryptionProtectorID(serverId.SubscriptionId, serverId.ResourceGroup, serverId.Name, "current")
+	d.SetId(id.ID())
+
+	return resourceMsSqlTransparentDataEncryptionRead(d, meta)
+}
+
+func resourceMsSqlTransparentDataEncryptionRead(d *schema.ResourceData, meta interface{}) error {
+	encryptionProtectorClient := meta.(*clients.Client).MSSQL.EncryptionProtectorClient
+
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.EncryptionProtectorID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := encryptionProtectorClient.Get(ctx, id.ResourceGroup, id.ServerName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request for Encryption Protector on server %s: %v", id.ServerName, err)
```

Comment:
Same here, we can use the id string representation

```suggestion
		return fmt.Errorf("making Read request for %s: %v", id, err)
```

Reply:
id is a struct here - Did you mean id.ID() or id.String()?